apps/server/src/modules/files-storage/controller/dto/file-storage.params.ts

```diff
@@ -1,9 +1,15 @@
 import { ApiProperty } from '@nestjs/swagger';
 import { EntityId } from '@shared/domain';
 import { FileRecordParentType } from '@shared/domain/entity/filerecord.entity';
-import { Allow, IsEnum, IsMongoId, IsNotEmpty, IsString, ValidateNested } from 'class-validator';
+import { Allow, IsBoolean, IsEnum, IsMongoId, IsNotEmpty, IsString, ValidateNested } from 'class-validator';
 
 export class FileRecordParams {
+	constructor(fileRecordParams: FileRecordParams) {
+		this.schoolId = fileRecordParams.schoolId;
+		this.parentId = fileRecordParams.parentId;
+		this.parentType = fileRecordParams.parentType;
+	}
+
```

Comment:
The Nestjs specific DTO classes should not have a constructor. Otherwise we have to look at this:
https://www.npmjs.com/package/class-transformer#plaintoclass

Reply:
Then let look into it togehter how we can arrived the instanceof without as cast.

https://github.com/hpi-schul-cloud/schulcloud-server/blob/BC-2476-filestorage-service-refactoring/apps/server/src/modules/files-storage/helper/file-storage.helper.spec.ts#L237